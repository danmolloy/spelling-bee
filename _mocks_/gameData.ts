import { faker } from "@faker-js/faker"

//const centerLetter = faker.string.alpha();
//const outerLetters = faker.string.alpha(6).split("")
const expiration = Date.now() + 10000
const answers = faker.word.words(25).toLowerCase().split("").filter(i => i !== " ")
const lettersArr = Array.from(new Set(answers)).slice(0, 7)

const validLetters = Array.from(new Set(faker.string.alpha(12).split(""))).slice(0, 7)

export const mockData = {
  expiration: expiration,
  displayWeekday:"Sunday",
  displayDate: "October 13, 1986",
  printDate:"2022-04-10",
  centerLetter: lettersArr[0],
  outerLetters: lettersArr.slice(1, 7),
  validLetters: lettersArr,
  pangrams: [lettersArr.join("")],
  answers: [lettersArr.join(""), lettersArr.slice(1, 6).join(""), ...faker.word.words(25).split(" ")],
  id: faker.string.uuid(),
  freeExpiration:"0",
  editor:"Sam Ezersky"
}