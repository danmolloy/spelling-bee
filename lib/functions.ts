export type GameData = {
  expiration: number,
  displayWeekday: string
  displayDate: string,
  printDate: string,
  centerLetter: string,
  outerLetters: string[],
  validLetters: string[],
  pangrams: string[],
  answers: string[],
  id: string,
  freeExpiration: string
  editor: string
}

export const handleSubmit = (word: string, gameData: GameData, foundWords: string[]): {
  message: string|null
  addedPoints: number
} => {
  const submitObj =  {
    message: "",
    addedPoints: 0,
  }
  if (word.length === 0) {
    submitObj.message = null
    submitObj.addedPoints = 0
  } else if (word.length < 4) {
    submitObj.message = "Too short"
    submitObj.addedPoints = 0
  } else if (!checkLetters(word, gameData.validLetters)) {
    submitObj.message = "Bad letters"
    submitObj.addedPoints = 0
  } else if (!gameData.answers.includes(word.toLowerCase())) {
    submitObj.message = "Not in word list"
    submitObj.addedPoints = 0
  } else if (foundWords.map(i => i.toLowerCase()).includes(word.toLowerCase())) {
    submitObj.message = "Already found"
    submitObj.addedPoints = 0
  } else if(
    !foundWords.map(i => i.toLowerCase()).includes(word.toLowerCase()) 
    && gameData.pangrams.includes(word.toLowerCase())) {
      submitObj.message = "Pangram!"
      submitObj.addedPoints = getPoints([word])
  } else if (
    !foundWords.map(i => i.toLowerCase()).includes(word.toLowerCase()) 
    && gameData.answers.map(i => i.toLowerCase()).includes(word.toLowerCase())) {
    const wordPoints = getPoints([word])
    submitObj.message = encouragement(wordPoints)
    submitObj.addedPoints = wordPoints
  }
  return submitObj;
}

export const encouragement = (wordPoints: number): string => {
    return wordPoints > 5 
    ? "Epic!"
    : wordPoints === 5 
    ? "Excellent!" 
    : wordPoints === 4 
    ? "Fantastic!"
    : wordPoints === 3 
    ? "Nice!"
    : wordPoints === 2 
    ? "Great!" 
    : "Good!"
}

export const checkLetters = (word: string, letters: string[]): boolean => {
  let wordArr: string[] = word.split("").map(i => i.toUpperCase())
  for (let i = 0; i < wordArr.length; i++) {
    if (!letters.map(i => i.toUpperCase()).includes(wordArr[i])) {
      return false
    }
  }
  return true
}

export const getPoints = (wordList: string[]) => {
  let sum = 0;
  if (wordList === undefined) {
    return sum
  }

  for (let i = 0; i < wordList?.length; i++) {
    if (wordList[i].length === 4) {
      sum += 1
    } else if (wordList[i].length > 4) {
      sum += wordList[i].length
    }
  }

	let word: Set<string>;
  for (let i = 0; i < wordList.length; i++) {
    word = new Set(wordList[i].split(''))
    if (word.size === 7) {
      sum += 7
    }
  }
  
  return sum;
}