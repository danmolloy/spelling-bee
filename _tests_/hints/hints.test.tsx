import "@testing-library/jest-dom"
import { screen, render, act, fireEvent } from "@testing-library/react"
import Hints, { HintsProps } from "../../components/hints"
import { mockData } from "../../_mocks_/gameData"
import { getPoints } from "../../lib/functions"

jest.spyOn(Storage.prototype, 'setItem')
Storage.prototype.setItem = jest.fn()

const mockProps: HintsProps = {
  centerLetter: mockData.centerLetter,
  validLetters: mockData.validLetters,
  setShowMenuItem: jest.fn(),
  answers: mockData.answers,
  foundWords: mockData.answers.slice(0, 5),
  pangrams: ['lorem'],
  //setRevealAnswers: jest.fn(),
  //revealAnswers: false
}

describe("<Hints />", () => {
  beforeEach(() => {
    render(<Hints {...mockProps} />)
  })
  it("hints-div is in the document", () => {
    const hintsDiv = screen.getByTestId("hints-div")
    expect(hintsDiv).toBeInTheDocument()
  })
  it("'Hints' title is in the document", () => {
    const hintsTitle = screen.getByText("Hints")
    expect(hintsTitle).toBeInTheDocument()
  })
  it("menu-icon is in the document and calls setShowMenuItem(null) on click", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    expect(menuIcon).toBeInTheDocument()
  })
  it("states number of pangrams", () => {
    const hintsDiv = screen.getByTestId("hints-div")
    expect(hintsDiv.textContent).toMatch(`Pangrams: ${mockProps.foundWords.filter(i => mockProps.pangrams.map(j => j.toLowerCase()).includes(i.toLowerCase())).length}/${mockData.pangrams.length}`)
  })
  it("states number of answers", () => {
    const hintsDiv = screen.getByTestId("hints-div")
    expect(hintsDiv.textContent).toMatch(`Words: ${mockProps.foundWords.length}/${mockData.answers.length}`)
  })

  it("states number of possible points", () => {
    const hintsDiv = screen.getByTestId("hints-div")
    expect(hintsDiv.textContent).toMatch(`Points: ${getPoints(mockProps.foundWords)}/${getPoints(mockData.answers)}`)
  })
  
  it("help-table is in the document", () => {
    const helpTable = screen.getByTestId("help-table")
    expect(helpTable).toBeInTheDocument()
  })

  it("two-letter list is in the document", () => {
    const twoLetterList = screen.getByTestId("two-letter-list")
    expect(twoLetterList).toBeInTheDocument()
  })
})

describe("<Hints />", () => {
  const mockProps: HintsProps = {
    centerLetter: "a",
    validLetters: ["q", "a", "b", "c", "d", "e", "f"],
    setShowMenuItem: jest.fn(),
    foundWords: mockData.answers.slice(0, 5),
    answers: ["four", "funf", "sixsix", "seventy", "eighty"],
    pangrams: ["lorem", "ipsum"]
  }
  beforeEach(() => {
    render(<Hints {...mockProps} />)
  })
  
  it("matches snapshot", () => {
    const hintsDiv = screen.getByTestId("hints-div")
    expect(hintsDiv).toMatchSnapshot()
  })
})