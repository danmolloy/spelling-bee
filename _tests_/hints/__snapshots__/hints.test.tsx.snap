// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hints /> matches snapshot 1`] = `
<div
  class="rounded mx-2 mt-24 p-4 pb-8 md:mx-24 lg:mx-60 shadow-md border flex flex-col  bg-white dark:bg-zinc-900 dark:text-zinc-300 dark:border dark:border-zinc-800"
  data-testid="hints-div"
>
  <div
    class="flex flex-row w-full justify-between items-center"
  >
    <h1
      class="text-2xl font-bold "
    >
      Hints
    </h1>
    <button
      class="hover:text-yellow-500 text-2xl m-2 w-10 h-10 rounded-full flex items-center justify-center"
      data-testid="menu-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </button>
  </div>
  <div
    class="p-2 font-light"
  >
    <p
      class=""
    >
      Words: 
      3/5
    </p>
    <p
      class=""
    >
      Pangrams: 0/2
    </p>
    <p
      class=""
    >
      Points: 
      8/21
    </p>
  </div>
  <div
    class="font-light flex flex-col items-center py-2 "
  >
    <div
      class="text-sm flex flex-row justify-between md:w-1/2 self-start md:self-auto "
    >
      <div
        class="flex  flex-row items-center"
        data-testid="letter-label"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
        <p
          class="ml-1"
        >
          Initial Letter
        </p>
      </div>
      <div
        class="flex flex-row items-center"
        data-testid="length-label"
      >
        <p
          class="mx-2"
        >
          Length of Word
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          />
        </svg>
      </div>
    </div>
    <table
      class="w-full md:w-1/2"
      data-testid="help-table"
    >
      <thead
        data-testid="table-head"
      >
        <tr
          class=" border-b"
        >
          <td
            class="w-10 h-10"
            data-testid="letter-col"
          />
          <td
            class="w-10 h-10"
          >
            <p
              class="text-center  w-full h-full flex items-center justify-center"
            >
              4
            </p>
          </td>
          <td
            class="w-10 h-10"
          >
            <p
              class="text-center  w-full h-full flex items-center justify-center"
            >
              6
            </p>
          </td>
          <td
            class="w-10 h-10"
          >
            <p
              class="text-center  w-full h-full flex items-center justify-center"
            >
              7
            </p>
          </td>
          <td
            class="w-10 h-10"
          >
            <p
              class="text-center w-full h-full flex items-center justify-center"
            >
              Total
            </p>
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="border-b"
          data-testid="q-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              Q
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-q"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-q"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-q"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="q-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <p>
                -
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="a-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              A
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-a"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-a"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-a"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="a-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <p>
                -
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="b-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              B
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-b"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-b"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-b"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="b-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <p>
                -
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="c-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              C
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-c"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-c"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-c"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="c-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <p>
                -
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="d-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              D
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-d"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-d"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-d"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="d-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <p>
                -
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="e-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              E
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-e"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-e"
          >
            <p
              class="font-bold bg-yellow-100 text-center w-full h-full flex items-center justify-center"
            >
              1
            </p>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-e"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="e-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              1
            </div>
          </td>
        </tr>
        <tr
          class="border-b"
          data-testid="f-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              F
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-f"
          >
            <div
              class="text-center text-bold text-4xl text-white w-full h-full bg-yellow-300  flex items-center justify-center"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
                />
              </svg>
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-f"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-f"
          >
            <div
              class=""
            />
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="f-total"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              <div
                class="text-center text-2xl text-black w-full h-full bg-yellow-300  flex items-center justify-center"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
        <tr
          data-testid="total-row"
        >
          <td
            class=" w-10 h-10"
          >
            <div
              class="text-center w-full h-full flex items-center justify-center"
            >
              Total
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="4-total"
          >
            <div
              class="text-center text-4xl text-white w-full h-full bg-yellow-300  flex items-center justify-center"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
                />
              </svg>
            </div>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="6-total"
          >
            <p
              class="text-center w-full h-full flex items-center justify-center"
            >
              1
            </p>
          </td>
          <td
            class="border-l w-10 h-10"
            data-testid="7-total"
          >
            <p
              class="text-center w-full h-full flex items-center justify-center"
            >
              1
            </p>
          </td>
          <td
            class="text-center border-l w-10 h-10"
            data-testid="total-words"
          >
            <p
              class="text-center w-full h-full flex items-center justify-center"
            >
              2
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="p-2 my-4"
    data-testid="two-letter-list"
  >
    <p
      class="font-bold"
    >
      Two-Letter List
    </p>
    <p
      class="my-2 text-sm font-light"
    >
      How many words remain for each pair of starting letters.
    </p>
    <div
      class="flex flex-row font-light"
      data-testid="e-row"
    >
      <div
        class=" flex flex-row border-b m-1 h-7  items-center"
        data-testid="EI-cell"
      >
        <p
          class="px-1"
        >
          EI
        </p>
        <div
          class="h-full w-7 flex items-center justify-center "
        >
          <p>
            1
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row font-light"
      data-testid="f-row"
    >
      <div
        class=" flex flex-row border-b m-1 h-7  items-center"
        data-testid="FO-cell"
      >
        <p
          class="px-1"
        >
          FO
        </p>
        <div
          class="h-full w-7 flex items-center justify-center "
        >
          <div
            class="text-center text-4xl text-white w-full h-full bg-yellow-300  flex items-center justify-center"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class=" flex flex-row border-b m-1 h-7  items-center"
        data-testid="FU-cell"
      >
        <p
          class="px-1"
        >
          FU
        </p>
        <div
          class="h-full w-7 flex items-center justify-center "
        >
          <div
            class="text-center text-4xl text-white w-full h-full bg-yellow-300  flex items-center justify-center"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row font-light"
      data-testid="s-row"
    >
      <div
        class=" flex flex-row border-b m-1 h-7  items-center"
        data-testid="SE-cell"
      >
        <p
          class="px-1"
        >
          SE
        </p>
        <div
          class="h-full w-7 flex items-center justify-center "
        >
          <p>
            1
          </p>
        </div>
      </div>
      <div
        class=" flex flex-row border-b m-1 h-7  items-center"
        data-testid="SI-cell"
      >
        <p
          class="px-1"
        >
          SI
        </p>
        <div
          class="h-full w-7 flex items-center justify-center "
        >
          <div
            class="text-center text-4xl text-white w-full h-full bg-yellow-300  flex items-center justify-center"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M362.6 192.9L345 174.8c-.7-.8-1.8-1.2-2.8-1.2-1.1 0-2.1.4-2.8 1.2l-122 122.9-44.4-44.4c-.8-.8-1.8-1.2-2.8-1.2-1 0-2 .4-2.8 1.2l-17.8 17.8c-1.6 1.6-1.6 4.1 0 5.7l56 56c3.6 3.6 8 5.7 11.7 5.7 5.3 0 9.9-3.9 11.6-5.5h.1l133.7-134.4c1.4-1.7 1.4-4.2-.1-5.7z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
